applications/StructuralMechanicsApplication/tests/cpp_tests/test_assign_nodal_elements_to_nodes_process.cpp

```diff
@@ -0,0 +1,120 @@
+        }
+
+        /**
+        * Checks the correct work of the AssignNodalElementsToNodesProcess process
+        * Test 1. No rotation
+        */
+        KRATOS_TEST_CASE_IN_SUITE(AssignNodalElementsToNodesProcess1, KratosStructuralMechanicsFastSuite)
+        {
+            Model model;
+            ModelPart& r_model_part = model.CreateModelPart("Main");
+
+            r_model_part.AddNodalSolutionStepVariable(DISPLACEMENT);
+            r_model_part.AddNodalSolutionStepVariable(VELOCITY);
+            r_model_part.AddNodalSolutionStepVariable(ACCELERATION);
+            AssignNodalElementsToNodesProcessCreateModelPart(r_model_part);
+
+            Parameters parameters = Parameters(R"(
+            {
+                "nodal_mass"                     : 1.0,
+                "nodal_stiffness"                : [1.0, null, 0.0]
+            })" );
+
+            AssignNodalElementsToNodesProcess assign_nodal_elements_to_nodes_process(r_model_part, parameters);
+            assign_nodal_elements_to_nodes_process.Execute();
+
+            KRATOS_CHECK_EQUAL(r_model_part.Elements().size(), r_model_part.Nodes().size());
+
+            for (auto& r_elem : r_model_part.Elements()) {
+                KRATOS_CHECK_EQUAL(r_elem.GetProperties()[NODAL_MASS], 1.0);
+                KRATOS_CHECK_EQUAL(r_elem.GetProperties()[NODAL_DISPLACEMENT_STIFFNESS][0], 1.0);
+                KRATOS_CHECK_EQUAL(r_elem.GetProperties()[NODAL_DISPLACEMENT_STIFFNESS][1], 0.0);
+                KRATOS_CHECK_EQUAL(r_elem.GetProperties()[NODAL_DISPLACEMENT_STIFFNESS][2], 0.0);
+            }
+        }
+
+        /**
+        * Checks the correct work of the AssignNodalElementsToNodesProcess process
+        * Test 2. WIth rotation
+        */
+        KRATOS_TEST_CASE_IN_SUITE(AssignNodalElementsToNodesProcess2, KratosStructuralMechanicsFastSuite)
+        {
+            Model model;
+            ModelPart& r_model_part = model.CreateModelPart("Main", 2);
+
+            r_model_part.AddNodalSolutionStepVariable(DISPLACEMENT);
+            r_model_part.AddNodalSolutionStepVariable(VELOCITY);
+            r_model_part.AddNodalSolutionStepVariable(ACCELERATION);
+            r_model_part.AddNodalSolutionStepVariable(ROTATION);
+            r_model_part.AddNodalSolutionStepVariable(ANGULAR_ACCELERATION);
+            r_model_part.AddNodalSolutionStepVariable(ANGULAR_ACCELERATION);
+            AssignNodalElementsToNodesProcessCreateModelPart(r_model_part);
+
+            Parameters parameters = Parameters(R"(
+            {
+                "nodal_mass"                     : 1.0,
+                "nodal_inertia"                  : [1.0, 1.0, 1.0],
+                "nodal_stiffness"                : [1.0, null, 0.0],
+                "nodal_rotational_stiffness"     : [1.0, null, 1.0]
+            })" );
+
+            AssignNodalElementsToNodesProcess assign_nodal_elements_to_nodes_process(r_model_part, parameters);
```

Comment:
does this work if you read directly from the mdpa without calling the process?

also, does it work from the gid interface?

Reply:
Yes, this is just because is easier and less rigid to work with processes

Reply:
It is not in the GiDInterface yet, but can easily be introduced